src/Elliptic/Systems/Poisson/Equations.hpp

```diff
@@ -16,6 +17,12 @@ class DataVector;
 namespace PUP {
 class er;
 }  // namespace PUP
+namespace Poisson {
+template <size_t Dim, Geometry BackgroundGeometry>
```

Comment:
Explain the ideas behind the Geometry template parameter.  What is it used for? (activate additional, more costly terms)  What are the requirements on it?  (it must be precisely one of the two tags specified in Elliptic/Systems/Poisson.hpp Geometry).   That kind of meta-information is very helpful to newcomers.   The generic forward declarations here are a good point to place such info, unless it's already elsewhere in a more coherent overview document.